src/task.rs

```diff
@@ -39,19 +39,19 @@ where
         .spawn_boxed(fut.boxed())
         .expect("cannot spawn a future");
 
-    JoinHandle { rx }
+    JoinHandleFuture { rx }
 }
 
 /// A handle that awaits the result of a [`spawn`]ed future.
 ///
 /// [`spawn`]: fn.spawn.html
 #[must_use = "futures do nothing unless you `.await` or poll them"]
 #[derive(Debug)]
-pub struct JoinHandle<T> {
+pub struct JoinHandleFuture<T> {
     pub(crate) rx: futures::channel::oneshot::Receiver<T>,
 }
 
-impl<T> Future for JoinHandle<T> {
+impl<T> Future for JoinHandleFuture<T> {
```

Comment:
Ah, this is probably the only special case; similar to `std::thread::JoinHandle` we should probably keep calling this `JoinHandle`.